packages/local_contexts/local_contexts.js

```diff
@@ -0,0 +1,169 @@
+                    </div>
+                </div>`
+    }
+
+    // console.log(project)
+    const notices = project.notice || []
+    const labels = (project.tk_labels || []).concat(project.bc_labels || [])
+    const lcWrapper = `<div style="max-width: 700px; margin-top: 10px;
+                border: 0.5px solid darkgray;border-radius: 2rem; background: white">
+        <div  style="padding: 12px;padding-bottom: 0; width: 100&; display: flex; flex-wrap: wrap; justify-content: flex-start">
+        ${notices.map(createItemImage).join("")}
+        ${labels.map(createItemImage).join("")}
+        </div>
+        <div style="padding: 12px; padding-top: 0; display: flex; flex-wrap: wrap; justify-content: flex-start; width: 100&">
+            <div style="display: inherit; margin: 0 8px">
+                <a href="https://localcontexts.org" target="_blank" style="display: inline-flex;">
+                    <img style="width:30px;" src="https://localcontexts.org/wp-content/uploads/2023/04/White-Background.png" alt="Local Contexts Icon">
+                </a>
+            </div>
+            <div style="align-items: center;display: flex;margin-right: 8px;font-size: 14px;">
+                <a id="project-link" style="font-weight: bold; color: #007585; cursor: pointer; font-size: 14px; text-decoration: underline;" href="${project.project_page}" target="_blank" rel="noopener noreferrer">${project.title}</a>
+            </div>
+        </div>
+        </div>`
+
+    const e = document.createElement("div")
+    e.innerHTML = lcWrapper
+    return e
+}
+
+async function cvoc_lc_LoadOrFetch(fullUrl) {
+    const lc_project_url_base = `${cvoc_lc_lcBaseUrl}/projects/`
+    let lc_uuid = fullUrl;
+    if (lc_uuid.startsWith(lc_project_url_base)) {
+        lc_uuid = lc_uuid.substring(lc_project_url_base.length);
+    }
+    const inStorage = sessionStorage.getItem(lc_uuid)
+    if (inStorage) {
+        return Promise.resolve(JSON.parse(inStorage))
+    }
+    const response = await fetch(`${cvoc_lc_lcBaseUrl}/api/v1/projects/${lc_uuid}`)
+    const project = await response.json()
+    sessionStorage.setItem(lc_uuid, JSON.stringify(project));
+    return Promise.resolve(project)
+}
+
+async function cvoc_lc_viewProject() {
+    // console.log("cvoc_lc_viewProject")
+    const jqSelect = $(cvoc_lc_projectSelector);
+    if (jqSelect.length === 0)
+        return
+    const projectField = jqSelect[0];
+    const fullUrl = projectField.textContent
+    // this hack needs to be done, when ready is executed twice. it prevents adding the LC container twice
+    // console.log(projectField.parentNode.children.length)
+    setTimeout(async () => {
+        if (projectField.parentNode.children.length === 1) {
+            const project = await cvoc_lc_LoadOrFetch(fullUrl);
+            let lcContainerElement = cvoc_lc_buildLCProjectPopup(project)
+            projectField.after(lcContainerElement);
+        }
+    }, Math.random() * 1000)
```

Comment:
FWIW - in other scripts, we add an 'expanded' attribute once we've processed a field so that it won't get changed twice - would that work here? Also - why Math.random()?

Reply:
That was a lot of experimenting and hacking. Math random to so there would be a time diff between checking if there is something to add... adding expanded asap seems to work well actually.